feat: Add new module `document-db/mongo-cluster`

zedy-wj opened the pull request:
## Description

Related PR: https://github.com/Azure/azure-dev/pull/3507

Add a new module `document-db/mongo-cluster` to support new feature.

<!--
>Thank you for your contribution !
> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

Fixes #123
Fixes #456
Closes #123
Closes #456
-->

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
|[![avm.res.document-db.mongo-cluster](https://github.com/zedy-wj/bicep-registry-modules/actions/workflows/avm.res.document-db.mongo-cluster.yml/badge.svg?branch=document-db%2FmongoClusters)](https://github.com/zedy-wj/bicep-registry-modules/actions/workflows/avm.res.document-db.mongo-cluster.yml) |

## Type of Change

<!-- Use the checkboxes [x] on the options that are relevant. -->

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/bicep -->

@jongio - for notification.

zedy-wj commented:
We are adding a new module to AVM: `document-db/mongo-clusters`. But we don't know exactly how to do that.

@AlexanderSehr - Sorry for ping you here, but my previous PRs were reviewed by you and I hope to get some help from you here, thanks in advance! I have tried to fix some errors in pipeline CI, but there are still three errors left:
```
Error: [-] [document-db/mongo-clusters] Telemetry deployment should have expected telemetry identifier. 269ms (268ms|1ms)
Error: [-] [document-db/mongo-clusters] Owning team should be specified correctly in CODEWONERS file. 12ms (11ms|1ms)
Error: [-] [document-db/mongo-clusters] Module identifier should be listed in issue template in the correct alphabetical position. 15ms (15ms|1ms)
```
Could you take a look and give me some guidance? Thanks a lot.

AlexanderSehr commented:
Hey @zedy-wj,
one hell of a contribution 🚀 
The answer is sadly as simple as it is crushing. You may or may not have bypassed the contribution process for new modules. 😉 And given that I don't find any module proposal for `document-db/mongo-clusters` [in AVM](https://github.com/Azure/Azure-Verified-Modules/issues), I'm fairly certain you did.
There are some manual steps we have to execute before we can publish anything into the Public Registry. To this end, the first step would be to open a new module proposal [here](https://github.com/Azure/Azure-Verified-Modules/issues/new?assignees=&labels=Type%3A+New+Module+Proposal+%3Abulb%3A%2CNeeds%3A+Triage+%3Amag%3A&projects=Azure%2F529&template=module_proposal.yml&title=%5BModule+Proposal%5D%3A+%60MODULE_NAME_BETWEEN_BACKTICKS%60).

The entire process is described [here](https://azure.github.io/Azure-Verified-Modules/contributing/process/). It's vital to follow the steps as without, publishing will fail, which will defeat the purpose of your PR 😄 

Please create the module proposal as linked above, and reach out to the core team as needed to accelerate the process. There are also a few other things like creating GitHub teams etc. (see contribution guide). 

In the meantime, I can already start reviewing the PR and only when we want to merge it we have to make sure that all other steps have been completed.

zedy-wj commented:
@AlexanderSehr - Thank you for your reply, I have submitted the [proposal](https://github.com/Azure/Azure-Verified-Modules/issues/1220).

@jongio - We have created a draft pr to migrate the features in [PR](https://github.com/Azure/azure-dev/pull/3507), since it involves the development of new module, we have created a proposal. Could you help review this [proposal](https://github.com/Azure/Azure-Verified-Modules/issues/1220)?

AlexanderSehr commented:
> @AlexanderSehr - Thank you for your reply, I have submitted the [proposal](https://github.com/Azure/Azure-Verified-Modules/issues/1220).
> 
> @jongio - We have created a draft pr to migrate the features in [PR](https://github.com/Azure/azure-dev/pull/3507), since it involves the development of new module, we have created a proposal. Could you help review this [proposal](https://github.com/Azure/Azure-Verified-Modules/issues/1220)?

Thanks @zedy-wj, beware that not 'owning' a proposed module may yield some problems as this means nobdoy would be immediately responsible for taking care of raised issues & PRs. Was that by intention?
I think the 'default' process when proposing a module without also taking ownership is to search for one before starting the implementation. @jtracey93 can you confirm?

zedy-wj commented:
> > @AlexanderSehr - Thank you for your reply, I have submitted the [proposal](https://github.com/Azure/Azure-Verified-Modules/issues/1220).
> > @jongio - We have created a draft pr to migrate the features in [PR](https://github.com/Azure/azure-dev/pull/3507), since it involves the development of new module, we have created a proposal. Could you help review this [proposal](https://github.com/Azure/Azure-Verified-Modules/issues/1220)?
> 
> Thanks @zedy-wj, beware that not 'owning' a proposed module may yield some problems as this means nobdoy would be immediately responsible for taking care of raised issues & PRs. Was that by intention? I think the 'default' process when proposing a module without also taking ownership is to search for one before starting the implementation. @jtracey93 can you confirm?

@jongio - Could you help confirm the owner for the new module.
@AlexanderSehr - Thanks for the reminder, the owner of the module needs jongio to confirm, let's wait for his reply first.

jongio commented:
Can we ask the owner of the cosmos module to take this as well?  @AlexanderSehr

AlexanderSehr commented:
> Can we ask the owner of the cosmos module to take this as well? @AlexanderSehr

Hey @jongio, you can certainly try and ask. The owners are all listed [here](https://azure.github.io/Azure-Verified-Modules/indexes/bicep/bicep-resource-modules), next to the module names. 
However, I wouldn't get my hopes up as the owner of Cosmos DB (@bryansan-msft) is pretty packed with work as it is, and it would likely not be in his interest to pick up even more if capacity doesn't allow it. But he'll know best 💪 
Is there a reason neither of you would want to own the module yourself? Is there also a capacity constraint? Naturally, you should only own a module if you can commit to maintain it, but I guess I have to ask 😄

jongio commented:
Hey @bryansan-msft - Would you be willing to take this one on too?  I was hoping it wouldn't be too much more work for you on top of the existing mongo/cosmos work.

bryansan-msft commented:
Hello @jongio to be honest I dont have the capacity to do this. I am taking too much time to do stuff in my current 3 modules. I have just orphaned the sql server one due to this.

FYI @AlexanderSehr

bryansan-msft commented:
Maybe we can release this initial version  and immediately orphan it? Similar to what we did with the old CARML modules

AlexanderSehr commented:
> Maybe we can release this initial version and immediately orphan it? Similar to what we did with the old CARML modules

If bad comes to worse, yes. In any case, I 100% support not picking up more than capacity allows. 💪

sinedied commented:
@AlexanderSehr what's the next step, now that I've been assigned ownership of this module?
I believe the technical core team can start reviewing the PR once it's ready now that the proposal has been accepted?

@zedy-wj What is currently missing in your PR to make it ready for review?

Thanks!

zedy-wj commented:
@sinedied - We will fix the issue in the pipeline as soon as possible to get it ready.

matebarabas commented:
> 3. @matebarabas must publish the module to an internal file that enables us to publish the module to the registry ❌

FYI, this is currently in PR. I'll let you know when it's merged, along with all the records of the 10 other AZD ptn modules.

zedy-wj commented:
Hey @sinedied, @AlexanderSehr - All requirements have been updated. PR is ready, please review it, thanks!

AlexanderSehr commented on .github/workflows/avm.res.document-db.mongo-cluster.yml:
Just a blocking comment until PR to MAR file is merged. No todo for you @zedy-wj

AlexanderSehr commented on avm/res/document-db/mongo-cluster/main.bicep:
```diff
@@ -0,0 +1,112 @@
+  name: '${uniqueString(deployment().name, location)}-configfwr'
+  params: {
+    mongoClusterName: mongoCluster.name
+    allowAllIPsFirewall: allowAllIPsFirewall
+    allowAzureIPsFirewall: allowAzureIPsFirewall
+    allowedSingleIPs: allowedSingleIPs
```
Just a design question: Shouldn't those be specified by the user, and we provide them as an example in the tests/readme, or a default value of sorts

AlexanderSehr replied:
@eriqua what's your take on this?

AlexanderSehr commented on avm/res/document-db/mongo-cluster/config-firewallrules/main.bicep:
Folder should be `firewall-rule`, as per the specs. Please take them to heart

AlexanderSehr commented on avm/res/document-db/mongo-cluster/config-firewallrules/main.bicep:
```diff
@@ -0,0 +1,51 @@
+@description('Optional. IP addresses to allow access to the cluster from.')
+param allowedSingleIPs array = []
+
+@description('Required. Name of the Mongo Cluster.')
```
```suggestion
@description('Conditional. The name of the parent mongo cluster. Required if the template is used in a standalone deployment.')
```

AlexanderSehr commented on avm/res/document-db/mongo-cluster/main.bicep:
```diff
@@ -0,0 +1,112 @@
+output resourceGroupName string = resourceGroup().name
+
+@description('The connection string key of the mongo cluster.')
+output connectionStringKey string = mongoCluster.properties.connectionString
```
Isn't this a credential/secret? If so, this output must be removed and can be replaced with the [secret export](https://azure.github.io/Azure-Verified-Modules/specs/bicep/interfaces/#secrets-export) pattern, if desired

AlexanderSehr commented on avm/res/document-db/mongo-cluster/tests/e2e/defaults/main.test.bicep:
```diff
@@ -0,0 +1,56 @@
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+// e.g., for a module 'network/private-endpoint' you could use 'dep-dev-network.privateendpoints-${serviceShort}-rg'
```
```suggestion
```

AlexanderSehr commented on avm/res/document-db/mongo-cluster/tests/e2e/defaults/main.test.bicep:
```diff
@@ -0,0 +1,56 @@
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+// e.g., for a module 'network/private-endpoint' you could use 'npe' as a prefix and then 'waf' as a suffix for the waf-aligned test
```
```suggestion
```

AlexanderSehr replied:
Applies to other test files too

AlexanderSehr commented on avm/res/document-db/mongo-cluster/tests/e2e/defaults/main.test.bicep:
```diff
@@ -0,0 +1,56 @@
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      // You parameters go here
```
```suggestion
```

AlexanderSehr replied:
Applies to other test files too

AlexanderSehr commented on avm/res/document-db/mongo-cluster/tests/e2e/defaults/main.test.bicep:
```diff
@@ -0,0 +1,56 @@
+      name: '${namePrefix}${serviceShort}001'
+      location: resourceLocation
+      administratorLogin: 'Admin001'
+      administratorLoginPassword: 'Admin123'
```
Please replace in this and other test files with an input parameter like here:
https://github.com/Azure/bicep-registry-modules/blob/c92a8c2bb229a777ae6812c06a09dfcb4fa28cb4/avm/res/compute/virtual-machine/tests/e2e/windows.defaults/main.test.bicep#L20-L22

AlexanderSehr commented on avm/res/document-db/mongo-cluster/main.bicep:
```diff
@@ -0,0 +1,112 @@
+  }
+}
+
+module configFireWallRules './config-firewallrules/main.bicep' = {
```
```suggestion
module mongoCluster_configFireWallRules './config-firewallrules/main.bicep' = {
```

AlexanderSehr commented on avm/res/document-db/mongo-cluster/config-firewallrules/main.bicep:
```diff
@@ -0,0 +1,51 @@
+  }
+]
+
+@description('The name of the resource group the mongo cluster was created in.')
+output resourceGroupName string = resourceGroup().name
```
```suggestion
@description('The name of the firewall rule.')
output name string = allowAllIPsFirewall ? firewall_all.name : allowAzureIPsFirewall ? firewall_azure.name : firewall_single.name

@description('The resource ID of the firewall rule.')
output resourceId string = allowAllIPsFirewall ? firewall_all.id : allowAzureIPsFirewall ? firewall_azure.id : firewall_single.id

@description('The name of the resource group the firewall rule was created in.')
output resourceGroupName string = resourceGroup().name
```

AlexanderSehr commented on avm/res/document-db/mongo-cluster/config-firewall-rule/main.bicep:
```diff
@@ -49,3 +49,36 @@ resource firewall_single 'Microsoft.DocumentDB/mongoClusters/firewallRules@2024-
+output firewallSingle firewallSetType[] = [
+  for index in range(0, length(allowedSingleIPs ?? [])): {
+    name: firewall_single[index].name
+    ResourceId: firewall_single[index].id
```
```suggestion
    resourceId: firewall_single[index].id
```

AlexanderSehr commented on avm/res/document-db/mongo-cluster/config-firewall-rule/main.bicep:
```diff
@@ -0,0 +1,84 @@
+  }
+}
+
+resource firewall_single 'Microsoft.DocumentDB/mongoClusters/firewallRules@2024-02-15-preview' = [
```
I only now see that this is a loop. Same as in my other comment, I'd strongly suggest to change the way this module is implemented. It's not aligned with the AVM spec [Single Resource Only](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-rmfr1---category-composition---single-resource-only) which is one of the mandatory ones.

How about generalizing this module to accept a start & end IP as per the resource type, and move all the logic / cases you implemented into the parent, that is 'mongo-cluster' where you could pass the intended parameters down to this module as defined. I'd still argue they should only be examples to make it easy for you, but you can of course implement it in a different way.

AlexanderSehr commented on avm/res/document-db/mongo-cluster/main.bicep:
Does the RP support RBAC, Diagnostic Settings and or Private Endpoints? If so, we should add their implementation as per the specs.

matebarabas commented:
@AlexanderSehr, @zedy-wj, good news: the MAR manifest has been updated with the required record(s) for this module. You can take the next steps accordingly. Thanks.

AlexanderSehr commented on avm/res/document-db/mongo-cluster/tests/e2e/max/main.test.bicep:
```diff
@@ -0,0 +1,71 @@
+        customRules: [
+          {
+            firewallRuleName: 'allow-1.2.3.4-to-5.6.7.8'
+            endIpAddress: '1.2.3.4'
+            startIpAddress: '5.6.7.8'
```
```suggestion
            startIpAddress: '1.2.3.4'
            endIpAddress: '5.6.7.8'
```

I guess it didn't like that (given the error in the workflow 😄 )

AlexanderSehr commented on avm/res/document-db/mongo-cluster/main.bicep:
```diff
@@ -0,0 +1,575 @@
+  'Key Vault Administrator': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '00482a5a-887f-4fb3-b363-3b7fe8e74483'
+  )
+  'Key Vault Certificates Officer': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'a4417e6f-fecd-4de8-b567-7b0420556985'
+  )
+  'Key Vault Certificate User': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'db79e9a7-68ee-4b58-9aeb-b90e7c24fcba'
+  )
+  'Key Vault Contributor': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'f25e0fa2-a7c8-4377-a976-54943a77a395'
+  )
+  'Key Vault Crypto Officer': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '14b46e9e-c2b7-41b4-b07b-48a6ebf60603'
+  )
+  'Key Vault Crypto Service Encryption User': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'e147488a-f6f5-4113-8e2d-b22465e65bf6'
+  )
+  'Key Vault Crypto User': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '12338af0-0e69-4776-bea7-57ae8d297424'
+  )
+  'Key Vault Reader': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '21090545-7ca7-4776-b22c-e363652d74d2'
+  )
+  'Key Vault Secrets Officer': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'b86a8fe4-44ce-4948-aee5-eccb2c155cd7'
+  )
+  'Key Vault Secrets User': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '4633458b-17de-408a-b874-0445c86b69e6'
+  )
```
Please replace those with roles relevant to the mongo-cluster resource

AlexanderSehr replied:
Just tried to find some via `Get-AzRoleDefinition`, but it does not appear there are any specific to this resource. If the case, just can just remove the KeyVault roles and leave the other 5 standard roles (Owner etc.) in

AlexanderSehr commented on avm/res/document-db/mongo-cluster/main.bicep:
```diff
@@ -0,0 +1,575 @@
+  @description('Required. The resource ID of the key vault where to store the secrets of this module.')
+  keyVaultResourceId: string
+
+  @description('Optional. The primary write connection string secret name to create.')
```
```suggestion
  @description('Optional. The name to use when creating the primary write connection string secret.')
```

AlexanderSehr commented on avm/res/document-db/mongo-cluster/tests/e2e/max/main.test.bicep:
```diff
@@ -0,0 +1,71 @@
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
```
As you added all of them, I'd recommend to copy tests for
- Private Endpoint
- RBAC
- Diagnostic Settings
to this test. You can find them in most other module's tests can copy them from there

AlexanderSehr commented on avm/res/document-db/mongo-cluster/main.bicep:
```diff
@@ -0,0 +1,575 @@
+  }
+}
+
+module mongoCluster_privateEndpoints 'br/public:avm/res/network/private-endpoint:0.7.1' = [
+  for (privateEndpoint, index) in (privateEndpoints ?? []): {
+    name: '${uniqueString(deployment().name, location)}-databaseAccount-PrivateEndpoint-${index}'
+    scope: resourceGroup(privateEndpoint.?resourceGroupName ?? '')
+    params: {
+      name: privateEndpoint.?name ?? 'pep-${last(split(mongoCluster.id, '/'))}-${privateEndpoint.service}-${index}'
+      privateLinkServiceConnections: privateEndpoint.?isManualConnection != true
+        ? [
+            {
+              name: privateEndpoint.?privateLinkServiceConnectionName ?? '${last(split(mongoCluster.id, '/'))}-${privateEndpoint.service}-${index}'
+              properties: {
+                privateLinkServiceId: mongoCluster.id
+                groupIds: [
+                  privateEndpoint.service
+                ]
+              }
+            }
+          ]
+        : null
+      manualPrivateLinkServiceConnections: privateEndpoint.?isManualConnection == true
+        ? [
+            {
+              name: privateEndpoint.?privateLinkServiceConnectionName ?? '${last(split(mongoCluster.id, '/'))}-${privateEndpoint.service}-${index}'
+              properties: {
+                privateLinkServiceId: mongoCluster.id
+                groupIds: [
+                  privateEndpoint.service
+                ]
+                requestMessage: privateEndpoint.?manualConnectionRequestMessage ?? 'Manual approval required.'
+              }
+            }
+          ]
+        : null
```
It seems only one value is available

```suggestion
module mongoCluster_privateEndpoints 'br/public:avm/res/network/private-endpoint:0.7.1' = [
  for (privateEndpoint, index) in (privateEndpoints ?? []): {
    name: '${uniqueString(deployment().name, location)}-databaseAccount-PrivateEndpoint-${index}'
    scope: resourceGroup(privateEndpoint.?resourceGroupName ?? '')
    params: {
      name: privateEndpoint.?name ?? 'pep-${last(split(mongoCluster.id, '/'))}-${privateEndpoint.?service ?? 'mongoCluster'}-${index}'
      privateLinkServiceConnections: privateEndpoint.?isManualConnection != true
        ? [
            {
              name: privateEndpoint.?privateLinkServiceConnectionName ?? '${last(split(mongoCluster.id, '/'))}-${privateEndpoint.?service ?? 'mongoCluster'}-${index}'
              properties: {
                privateLinkServiceId: mongoCluster.id
                groupIds: [
                  privateEndpoint.?service ?? 'mongoCluster'
                ]
              }
            }
          ]
        : null
      manualPrivateLinkServiceConnections: privateEndpoint.?isManualConnection == true
        ? [
            {
              name: privateEndpoint.?privateLinkServiceConnectionName ?? '${last(split(mongoCluster.id, '/'))}-${privateEndpoint.?service ?? 'mongoCluster'}-${index}'
              properties: {
                privateLinkServiceId: mongoCluster.id
                groupIds: [
                  privateEndpoint.?service ?? 'mongoCluster'
                ]
                requestMessage: privateEndpoint.?manualConnectionRequestMessage ?? 'Manual approval required.'
              }
            }
          ]
        : null
```

AlexanderSehr commented on avm/res/document-db/mongo-cluster/main.bicep:
```diff
@@ -0,0 +1,575 @@
+  @description('Optional. The name of the private link connection to create.')
+  privateLinkServiceConnectionName: string?
+
+  @description('Required. The subresource to deploy the private endpoint for. For example "blob", "table", "queue" or "file".')
+  service: string
```
Linked to https://github.com/Azure/bicep-registry-modules/pull/2790/files#r1750141770

```suggestion
  @description('Optional. The subresource to deploy the private endpoint for. For example "vault", "mysqlServer" or "dataFactory".')
    service: string?
```

AlexanderSehr commented on avm/res/document-db/mongo-cluster/tests/e2e/max/main.test.bicep:
```diff
@@ -0,0 +1,163 @@
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-diagnosticDependencies'
+  params: {
+    storageAccountName: 'dep${namePrefix}diasa${serviceShort}01'
```
```suggestion
    storageAccountName: 'dep${namePrefix}diasa${serviceShort}'
```

Maybe this helps the failed test

AlexanderSehr commented:
Great work @zedy-wj,
I think we're pretty much ready to go. Just that quota issue is a bit concerning - and will probably hit us the same way it did you

zedy-wj commented:
> Great work @zedy-wj 🚀, I think we're pretty much ready to go. Just that quota issue is a bit concerning - and will probably hit us the same way it did you 👀 Before we merge the PR in, we should make sure it works and take the necessary steps in both your, as well as the upstream environment so that the tests will pass.

Hey @AlexanderSehr,
Thank you for your quick review. I have updated the test location and the tests can now pass normally. Please re-review it, thanks!

AlexanderSehr commented on .github/CODEOWNERS:
```diff
@@ -62,6 +62,7 @@
 /avm/res/dev-test-lab/lab/ @Azure/avm-res-devtestlab-lab-module-owners-bicep @Azure/avm-module-reviewers-bicep
 /avm/res/digital-twins/digital-twins-instance/ @Azure/avm-res-digitaltwins-digitaltwinsinstance-module-owners-bicep @Azure/avm-module-reviewers-bicep
 /avm/res/document-db/database-account/ @Azure/avm-res-documentdb-databaseaccount-module-owners-bicep @Azure/avm-module-reviewers-bicep
+/avm/res/document-db/mongo-cluster/ @Azure/avm-res-documentdb-mongocluster-module-owners-bicep @Azure/avm-module-reviewers-bicep
```
Hey @zedy-wj,
just noticed this file shows errors. The reason is usually that the team in question (in this case `Azure/avm-res-documentdb-mongocluster-module-owners-bicep`) is not setup, or not setup correctly. Could you please check if that team is setup & is placed correct in the hierachy? This is required so that those configured as code-owners can subsequently approve PRs without the need to involve the repository maintainers. The corresponding spec with instructions can be found here: [SNFR20 - Category: Contribution/Support - GitHub Teams Only](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only).

zedy-wj replied:
@sinedied - Could you please follow the [instructions](https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-snfr20---category-contributionsupport---github-teams-only) and create a team named: `avm-res-documentdb-mongocluster-module-owners-bicep` since I'm not authorized (Non-FTE), I can't do that.

sinedied replied:
@zedy-wj All the required teams have been created, it should be good now

zedy-wj replied:
Hey @AlexanderSehr,
At present, I can indeed check that the team: `avm-res-documentdb-mongocluster-module-owners-bicep` has been setup, but this file still reports an error. Does the owner need any additional steps?

sinedied replied:
@zedy-wj after looking again, it seems it was missing the parent teams addition to get the permissions right, so I added this. Access will have to be granted by the parent teams, then it should work

matebarabas replied:
@sinedied, I've just approved these team requests.

AlexanderSehr replied:
Just came back for some time off. Looks good & will resolve

zedy-wj commented:
@AlexanderSehr, @sinedied - We have updated according to the comments, PR is ready. Please re-review it, thanks!

sinedied commented on avm/res/document-db/mongo-cluster/modules/keyVaultExport.bicep:
```diff
@@ -0,0 +1,50 @@
+@description('Required. The name of the Key Vault to set the ecrets in.')
```
```suggestion
@description('Required. The name of the Key Vault to set the secrets in.')
```

sinedied reviewed (commented):
Found a small typo.

Also I've seen that you mention "DocumentDB" and "Mongo cluster" in the documention, shoudn't we instead use the official commercial name of the service, ie **Cosmos DB for MongoDB vCore**?

See https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/

zedy-wj commented:
> Found a small typo.
> 
> Also I've seen that you mention "DocumentDB" and "Mongo cluster" in the documention, shoudn't we instead use the official commercial name of the service, ie **Cosmos DB for MongoDB vCore**?
> 
> See https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/

Updated to the official commercial name of the service, and fixed typos. Thanks for your review.

matebarabas commented:
> Found a small typo.
> 
> Also I've seen that you mention "DocumentDB" and "Mongo cluster" in the documention, shoudn't we instead use the official commercial name of the service, ie **Cosmos DB for MongoDB vCore**?
> 
> See https://learn.microsoft.com/en-us/azure/cosmos-db/mongodb/vcore/

Good point, @sinedied! I'll also make sure this gets accurately reflected in the module index, when it's updated next. Thank you for calling it out!
